### ISSUE TITLE
Not able to pass a third party type to function for Blazorworker

### ISSUE THREAD

**Issue by bpyzik** [Tewr/BlazorWorker#27]
I am running into the issue of passing a file type into the function in runasync as shown here: 

```
var service = await worker.CreateBackgroundServiceAsync<ExcelService>(
			  options => options
			  .AddConventionalAssemblyOfService()
			  .AddAssemblies("BlazorInputFile.dll")
			  .AddAssemblies("DocumentFormat.OpenXml.dll")
			  .AddAssemblies("DocumentFormat.OpenXml.Packaging.dll")
			  .AddAssemblies("DocumentFormat.OpenXml.Spreadsheet.dll")
			  .AddAssemblies("System.IO.Packaging.dll")
			  .AddAssemblies("System.Xml.Linq.dll")
			);
var result = await service.RunAsync(s => s.GetLinesFromSheet(files));
```
and I recieve error:
```
System.Runtime.Serialization.SerializationException: Error converting type: Type 'BlazorInputFile.FileListEntryImpl[]' with data contract name 'ArrayOfFileListEntryImpl:http://schemas.datacontract.org/2004/07/BlazorInputFile' is not expected.`
```

files is of type IFileEntry[] from the BlazorInputFile package. Is there a way to let the serviceworker know how to serialize this?

**Comment by Tewr** [comment]
Maybe, but far from simple.

The serialization of the expression depends on the default expression serializer: https://github.com/Tewr/BlazorWorker/blob/master/src/BlazorWorker.WorkerBackgroundService/SerializeLinqExpressionSerializer.cs

There is no simple way of replacing this serializer with you own implementation, nor configuring serialization for a specific type... I had plans to make it configurable somehow, however the intention has never become practice. It's difficult to design a simple API around this stuff and also pass it over the process border.

Id suggest serializing any unsupported type by yourself and pass it over using a byte array or a string.

**Comment by Tewr** [comment]
@bpyzik  Do you mean IFileListEntry ? https://github.com/SteveSandersonMS/BlazorInputFile/blob/master/BlazorInputFile/IFileListEntry.cs

That interface has a stream reference. Streams are, by their nature, not serializable. There is [an issue](https://github.com/Tewr/BlazorWorker/issues/3) open for supporting streams though.

**Comment by Matheos96** [comment]
> Id suggest serializing any unsupported type by yourself and pass it over using a byte array or a string.

Scrolling issues during the implementation phase and found this comment. Very useful tip as I was having issues passing semi-complex objects to the workers. We have our own serialization to BSON (Binary json) configured so I decided to just use that manually on the main thread and worker side for serialization and deserialization. I tried going down the rabbit hole of getting the built in serialization to work with our complex Data Model but it ended up getting harder and harder and eventually started messing with our own serialization functions...
